src/main/java/sparksoniq/jsoniq/runtime/tupleiterator/RuntimeTupleIterator.java

```diff
@@ -96,5 +101,44 @@ public IteratorMetadata getMetadata() {
 
     public abstract JavaRDD<FlworTuple> getRDD(DynamicContext context);
 
+    public Set<String> getVariableDependencies()
+    {
+        Set<String> result = new HashSet<String>();
+        result.addAll(_child.getVariableDependencies());
+        return result;
+    }
 
+    public Set<String> getBoundVariables()
```

Comment:
I had the intuition that BoundVariables here meant FLWOR variables and your recent addition of print function also confirms that with its style. However, my initial worry was, what if variables were defined somewhere else? This would cause a bit of confusion since the function's name suggest it gets all variable bindings in the application. Even though we don't  support it at the moment, I have noticed global variable definitions can be implemented further down the road (http://www.jsoniq.org/docs/JSONiq/html-single/index.html#idm16811680).

I would recommend indicating that BoundVariables here are actually BoundFlworVariables to bring more clarity and ease of extensibility down the road.